tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1270 @@
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "X_encode = torch.randn(batch_size, to_len, to_dim)\n",
+    "X_decode = torch.randn(batch_size, from_len, from_dim)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "53ba793f-af3d-4e09-89a2-72b3f8964158",
+   "metadata": {},
+   "source": [
+    "#### Init of models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "f3c2fd50-bc9b-4911-bd7c-8f8f03bd4ea4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 41.6 ms, sys: 1.91 ms, total: 43.5 ms\n",
+      "Wall time: 42.6 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "tdtransformer = TransformerTensorDict(\n",
+    "    num_blocks,\n",
+    "    \"X_encode\",\n",
+    "    \"X_decode\",\n",
+    "    to_dim,\n",
+    "    to_len,\n",
+    "    from_dim,\n",
+    "    from_len,\n",
+    "    latent_dim,\n",
+    "    num_heads,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "dfbadd6b-7847-4399-9b22-7e5c58524334",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 6.14 ms, sys: 1.75 ms, total: 7.89 ms\n",
+      "Wall time: 6.68 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "transformer = Transformer(\n",
+    "                num_blocks,\n",
+    "                to_dim,\n",
+    "                to_len,\n",
+    "                from_dim,\n",
+    "                from_len,\n",
+    "                latent_dim,\n",
+    "                num_heads\n",
+    "            )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6a63de8f-ee8e-4ddf-bf89-f72c2896e1c3",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "#### Inference time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "02a4116b-2b75-47fc-8bc1-3903aa7cd504",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 7.72 ms, sys: 8.6 ms, total: 16.3 ms\n",
+      "Wall time: 9.65 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "tokens = tdtransformer(td_tokens)\n",
+    "#"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "40158aab-b53a-4a99-82cb-f5595eef7159",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 6.02 ms, sys: 9.13 ms, total: 15.2 ms\n",
+      "Wall time: 7.83 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "X_out = transformer(X_encode, X_decode)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "664adff3-1466-47c3-9a80-a0f26171addd",
+   "metadata": {},
+   "source": [
+    "We can see on this minimal example that the overhead introduced by `TensorDictModule` is minimal."
```

Comment:
> introduced by `TensorDictModule` is minimal.

to
> introduced by `TensorDictModule` is marginal.

Also, I would remove the init time (where TensorDictModule clearly underperforms), it's irrelevant (of course it takes more time, but it's not an operation we will repeat often)

After this we're good to go